test/array.cpp

```diff
@@ -341,7 +341,10 @@ TEST(array_tricky_copy) {
 
 TEST(array_for_each_value_scalar) {
   array_of_rank<int, 0> scalar;
-  scalar.for_each_value([](int& v) { v = 3; });
+  scalar.for_each_value([token = no_copy{no_copy()}](int& v) {
```

Comment:
It might be more explicit if you did:

```
no_copy token;
scalar.for_each_value([token = std::move(token)] ...
```

(`token = std::move(token)` might be a little weird too. Feel free to rename either the outer or inner but it's perfectly valid if you think about it).